File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
import sys
from collections import deque
input = sys.stdin.readline

V, E = map(int, input().split())
adj = [[] for _ in range(V+1)]
for _ in range(E):
    a, b = map(int, input().split())
    adj[a].append(b)
    adj[b].append(a)
    
visited = [False] * (V+1)

def bfs(start):
    que = deque([start])
    visited[start] = True
    while que:
        v = que.popleft()
        for i in adj[v]:
            if not visited[i]:
                que.append(i)
                visited[i]=True

count =0
for i in range(1,V+1):
    if visited[i] == False:
        bfs(i)
        count+=1
        
print(count)

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.js
// let input = require('fs').readFileSync('11724.txt').toString().split('\n');
let input = require('fs').readFileSync('dev/stdin').toString().split('\n');

const [node, edge] = input[0].split(' ').map((v) => Number(v));

let graph = [];
for (let i = 0; i <= node; i++) {
  graph.push([]);
}

for (let i = 0; i < edge; i++) {
  const [a, b] = input[i + 1].split(' ').map((v) => Number(v));
  graph[a].push(b);
  graph[b].push(a);
}

let visited_bfs = new Array(node + 1).fill(false);
function bfs(graph, start) {
  let queue = [start];
  visited_bfs[start] = true;
  while (queue.length) {
    let v = queue.shift();
    for (let i = 0; i < graph[v].length; i++) {
      if (visited_bfs[graph[v][i]] === false) {
        queue.push(graph[v][i]);
        visited_bfs[graph[v][i]] = true;
      }
    }
  }
}

let count = 0;
for (let i = 1; i <= node; i++) {
  if (!visited_bfs[i]) {
    bfs(graph, i);
    count += 1;
  }
}

console.log(count);
